{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "eb8cf439-2b67-46fb-ae16-ccca2a3d26a9",
   "metadata": {},
   "outputs": [],
   "source": [
    "#flu selections"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "ec7aaa6d-6d80-4c61-ab6a-1cc5f993060c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "pd.set_option('display.max_rows', 10)\n",
    "pd.set_option('display.max_columns', None)\n",
    "\n",
    "import os\n",
    "import matplotlib\n",
    "import seaborn as sns\n",
    "\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "\n",
    "import warnings\n",
    "import itertools\n",
    "import numpy as np\n",
    "\n",
    "import tqdm\n",
    "import gc\n",
    "\n",
    "import scipy as sp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e4d80512-a770-4939-8b5f-dddf8c6ca38f",
   "metadata": {},
   "outputs": [],
   "source": [
    "warnings.simplefilter('ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "689e3b47-53bf-4e22-8139-39538bb28625",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>virus_background</th>\n",
       "      <th>replicate</th>\n",
       "      <th>selection</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>Lane</th>\n",
       "      <th>inline_fwd</th>\n",
       "      <th>inline_rev</th>\n",
       "      <th>read_1_file</th>\n",
       "      <th>read_2_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>USSR77_R1_Plasmid_L1</td>\n",
       "      <td>USSR77</td>\n",
       "      <td>R1</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>TTGATCG</td>\n",
       "      <td>TACCTGA</td>\n",
       "      <td>data/sequencing_data/USSR77_R1_Plasmid_S1_L001...</td>\n",
       "      <td>data/sequencing_data/USSR77_R1_Plasmid_S1_L001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>USSR77_R1_Plasmid_L2</td>\n",
       "      <td>USSR77</td>\n",
       "      <td>R1</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>TTGATCG</td>\n",
       "      <td>TACCTGA</td>\n",
       "      <td>data/sequencing_data/USSR77_R1_Plasmid_S1_L002...</td>\n",
       "      <td>data/sequencing_data/USSR77_R1_Plasmid_S1_L002...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>USSR77_R2_Plasmid_L1</td>\n",
       "      <td>USSR77</td>\n",
       "      <td>R2</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>CATTCTCGAG</td>\n",
       "      <td>AGAACCAT</td>\n",
       "      <td>data/sequencing_data/USSR77_R2_Plasmid_S2_L001...</td>\n",
       "      <td>data/sequencing_data/USSR77_R2_Plasmid_S2_L001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>USSR77_R2_Plasmid_L2</td>\n",
       "      <td>USSR77</td>\n",
       "      <td>R2</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>CATTCTCGAG</td>\n",
       "      <td>AGAACCAT</td>\n",
       "      <td>data/sequencing_data/USSR77_R2_Plasmid_S2_L002...</td>\n",
       "      <td>data/sequencing_data/USSR77_R2_Plasmid_S2_L002...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>USSR77_R1_P1_L1</td>\n",
       "      <td>USSR77</td>\n",
       "      <td>R1</td>\n",
       "      <td>P1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>CCTTAGTACTA</td>\n",
       "      <td>ACCGTTGAC</td>\n",
       "      <td>data/sequencing_data/USSR77_R1_P1_S3_L001_R1_0...</td>\n",
       "      <td>data/sequencing_data/USSR77_R1_P1_S3_L001_R2_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>SYD21_R2_Plasmid_L2</td>\n",
       "      <td>SYD21</td>\n",
       "      <td>R2</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td>TTGATCG</td>\n",
       "      <td>AGAACCAT</td>\n",
       "      <td>data/sequencing_data/SYD21_R2_Plasmid_S14_L002...</td>\n",
       "      <td>data/sequencing_data/SYD21_R2_Plasmid_S14_L002...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>SYD21_R1_P1_L1</td>\n",
       "      <td>SYD21</td>\n",
       "      <td>R1</td>\n",
       "      <td>P1</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>CATTCTCGAG</td>\n",
       "      <td>ACCGTTGAC</td>\n",
       "      <td>data/sequencing_data/SYD21_R1_P1_S15_L001_R1_0...</td>\n",
       "      <td>data/sequencing_data/SYD21_R1_P1_S15_L001_R2_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>SYD21_R1_P1_L2</td>\n",
       "      <td>SYD21</td>\n",
       "      <td>R1</td>\n",
       "      <td>P1</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>CATTCTCGAG</td>\n",
       "      <td>ACCGTTGAC</td>\n",
       "      <td>data/sequencing_data/SYD21_R1_P1_S15_L002_R1_0...</td>\n",
       "      <td>data/sequencing_data/SYD21_R1_P1_S15_L002_R2_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>SYD21_R2_P1_L1</td>\n",
       "      <td>SYD21</td>\n",
       "      <td>R2</td>\n",
       "      <td>P1</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>CCTTAGTACTA</td>\n",
       "      <td>GCAAGCTGTT</td>\n",
       "      <td>data/sequencing_data/SYD21_R2_P1_S16_L001_R1_0...</td>\n",
       "      <td>data/sequencing_data/SYD21_R2_P1_S16_L001_R2_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>SYD21_R2_P1_L2</td>\n",
       "      <td>SYD21</td>\n",
       "      <td>R2</td>\n",
       "      <td>P1</td>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>CCTTAGTACTA</td>\n",
       "      <td>GCAAGCTGTT</td>\n",
       "      <td>data/sequencing_data/SYD21_R2_P1_S16_L002_R1_0...</td>\n",
       "      <td>data/sequencing_data/SYD21_R2_P1_S16_L002_R2_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name virus_background replicate selection  sample_id  \\\n",
       "0   USSR77_R1_Plasmid_L1           USSR77        R1   Plasmid          1   \n",
       "1   USSR77_R1_Plasmid_L2           USSR77        R1   Plasmid          1   \n",
       "2   USSR77_R2_Plasmid_L1           USSR77        R2   Plasmid          2   \n",
       "3   USSR77_R2_Plasmid_L2           USSR77        R2   Plasmid          2   \n",
       "4        USSR77_R1_P1_L1           USSR77        R1        P1          3   \n",
       "..                   ...              ...       ...       ...        ...   \n",
       "27   SYD21_R2_Plasmid_L2            SYD21        R2   Plasmid         14   \n",
       "28        SYD21_R1_P1_L1            SYD21        R1        P1         15   \n",
       "29        SYD21_R1_P1_L2            SYD21        R1        P1         15   \n",
       "30        SYD21_R2_P1_L1            SYD21        R2        P1         16   \n",
       "31        SYD21_R2_P1_L2            SYD21        R2        P1         16   \n",
       "\n",
       "    Lane   inline_fwd  inline_rev  \\\n",
       "0      1      TTGATCG     TACCTGA   \n",
       "1      2      TTGATCG     TACCTGA   \n",
       "2      1   CATTCTCGAG    AGAACCAT   \n",
       "3      2   CATTCTCGAG    AGAACCAT   \n",
       "4      1  CCTTAGTACTA   ACCGTTGAC   \n",
       "..   ...          ...         ...   \n",
       "27     2      TTGATCG    AGAACCAT   \n",
       "28     1   CATTCTCGAG   ACCGTTGAC   \n",
       "29     2   CATTCTCGAG   ACCGTTGAC   \n",
       "30     1  CCTTAGTACTA  GCAAGCTGTT   \n",
       "31     2  CCTTAGTACTA  GCAAGCTGTT   \n",
       "\n",
       "                                          read_1_file  \\\n",
       "0   data/sequencing_data/USSR77_R1_Plasmid_S1_L001...   \n",
       "1   data/sequencing_data/USSR77_R1_Plasmid_S1_L002...   \n",
       "2   data/sequencing_data/USSR77_R2_Plasmid_S2_L001...   \n",
       "3   data/sequencing_data/USSR77_R2_Plasmid_S2_L002...   \n",
       "4   data/sequencing_data/USSR77_R1_P1_S3_L001_R1_0...   \n",
       "..                                                ...   \n",
       "27  data/sequencing_data/SYD21_R2_Plasmid_S14_L002...   \n",
       "28  data/sequencing_data/SYD21_R1_P1_S15_L001_R1_0...   \n",
       "29  data/sequencing_data/SYD21_R1_P1_S15_L002_R1_0...   \n",
       "30  data/sequencing_data/SYD21_R2_P1_S16_L001_R1_0...   \n",
       "31  data/sequencing_data/SYD21_R2_P1_S16_L002_R1_0...   \n",
       "\n",
       "                                          read_2_file  \n",
       "0   data/sequencing_data/USSR77_R1_Plasmid_S1_L001...  \n",
       "1   data/sequencing_data/USSR77_R1_Plasmid_S1_L002...  \n",
       "2   data/sequencing_data/USSR77_R2_Plasmid_S2_L001...  \n",
       "3   data/sequencing_data/USSR77_R2_Plasmid_S2_L002...  \n",
       "4   data/sequencing_data/USSR77_R1_P1_S3_L001_R2_0...  \n",
       "..                                                ...  \n",
       "27  data/sequencing_data/SYD21_R2_Plasmid_S14_L002...  \n",
       "28  data/sequencing_data/SYD21_R1_P1_S15_L001_R2_0...  \n",
       "29  data/sequencing_data/SYD21_R1_P1_S15_L002_R2_0...  \n",
       "30  data/sequencing_data/SYD21_R2_P1_S16_L001_R2_0...  \n",
       "31  data/sequencing_data/SYD21_R2_P1_S16_L002_R2_0...  \n",
       "\n",
       "[32 rows x 10 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples=pd.read_csv('data/samples.csv')\n",
    "samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 450,
   "id": "9f5f35ef-27c8-486f-a367-81131b62bd27",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/USSR77_R1_Plasmid_S1_L001_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/USSR77_R1_Plasmid_S1_L001_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.253568\n",
      "  C: 0.246511\n",
      "  G: 0.247004\n",
      "  T: 0.252917\n",
      "  42707 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 17,891,620 / 18,169,531 (98.470%)\n",
      "Discarded reads ...................: 0 / 18,169,531 (0.000%)\n",
      "Not assembled reads ...............: 277,911 / 18,169,531 (1.530%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/USSR77_R1_Plasmid_L1.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/USSR77_R1_Plasmid_L1.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/USSR77_R1_Plasmid_L1.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/USSR77_R1_Plasmid_L1.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/USSR77_R1_Plasmid_S1_L002_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/USSR77_R1_Plasmid_S1_L002_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.253497\n",
      "  C: 0.246557\n",
      "  G: 0.246987\n",
      "  T: 0.252959\n",
      "  42070 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 18,481,616 / 18,769,171 (98.468%)\n",
      "Discarded reads ...................: 0 / 18,769,171 (0.000%)\n",
      "Not assembled reads ...............: 287,555 / 18,769,171 (1.532%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/USSR77_R1_Plasmid_L2.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/USSR77_R1_Plasmid_L2.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/USSR77_R1_Plasmid_L2.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/USSR77_R1_Plasmid_L2.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/USSR77_R2_Plasmid_S2_L001_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/USSR77_R2_Plasmid_S2_L001_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.255346\n",
      "  C: 0.249634\n",
      "  G: 0.249142\n",
      "  T: 0.245878\n",
      "  45972 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 18,553,528 / 19,000,587 (97.647%)\n",
      "Discarded reads ...................: 0 / 19,000,587 (0.000%)\n",
      "Not assembled reads ...............: 447,059 / 19,000,587 (2.353%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/USSR77_R2_Plasmid_L1.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/USSR77_R2_Plasmid_L1.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/USSR77_R2_Plasmid_L1.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/USSR77_R2_Plasmid_L1.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/USSR77_R2_Plasmid_S2_L002_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/USSR77_R2_Plasmid_S2_L002_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.255211\n",
      "  C: 0.249723\n",
      "  G: 0.249171\n",
      "  T: 0.245895\n",
      "  43146 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 19,202,185 / 19,668,933 (97.627%)\n",
      "Discarded reads ...................: 0 / 19,668,933 (0.000%)\n",
      "Not assembled reads ...............: 466,748 / 19,668,933 (2.373%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/USSR77_R2_Plasmid_L2.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/USSR77_R2_Plasmid_L2.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/USSR77_R2_Plasmid_L2.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/USSR77_R2_Plasmid_L2.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/USSR77_R1_P1_S3_L001_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/USSR77_R1_P1_S3_L001_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.250936\n",
      "  C: 0.253873\n",
      "  G: 0.240443\n",
      "  T: 0.254748\n",
      "  64796 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 27,430,342 / 27,738,609 (98.889%)\n",
      "Discarded reads ...................: 0 / 27,738,609 (0.000%)\n",
      "Not assembled reads ...............: 308,267 / 27,738,609 (1.111%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/USSR77_R1_P1_L1.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/USSR77_R1_P1_L1.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/USSR77_R1_P1_L1.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/USSR77_R1_P1_L1.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/USSR77_R1_P1_S3_L002_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/USSR77_R1_P1_S3_L002_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.250840\n",
      "  C: 0.253938\n",
      "  G: 0.240473\n",
      "  T: 0.254749\n",
      "  60303 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 27,162,240 / 27,467,729 (98.888%)\n",
      "Discarded reads ...................: 0 / 27,467,729 (0.000%)\n",
      "Not assembled reads ...............: 305,489 / 27,467,729 (1.112%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/USSR77_R1_P1_L2.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/USSR77_R1_P1_L2.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/USSR77_R1_P1_L2.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/USSR77_R1_P1_L2.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/USSR77_R2_P1_S4_L001_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/USSR77_R2_P1_S4_L001_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.262029\n",
      "  C: 0.238324\n",
      "  G: 0.253395\n",
      "  T: 0.246253\n",
      "  52427 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 21,998,618 / 22,050,602 (99.764%)\n",
      "Discarded reads ...................: 0 / 22,050,602 (0.000%)\n",
      "Not assembled reads ...............: 51,984 / 22,050,602 (0.236%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/USSR77_R2_P1_L1.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/USSR77_R2_P1_L1.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/USSR77_R2_P1_L1.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/USSR77_R2_P1_L1.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/USSR77_R2_P1_S4_L002_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/USSR77_R2_P1_S4_L002_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.261960\n",
      "  C: 0.238359\n",
      "  G: 0.253437\n",
      "  T: 0.246244\n",
      "  49391 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 21,895,884 / 21,947,733 (99.764%)\n",
      "Discarded reads ...................: 0 / 21,947,733 (0.000%)\n",
      "Not assembled reads ...............: 51,849 / 21,947,733 (0.236%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/USSR77_R2_P1_L2.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/USSR77_R2_P1_L2.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/USSR77_R2_P1_L2.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/USSR77_R2_P1_L2.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SN89_R1_Plasmid_S5_L001_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SN89_R1_Plasmid_S5_L001_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.246685\n",
      "  C: 0.244651\n",
      "  G: 0.254023\n",
      "  T: 0.254640\n",
      "  52340 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 22,170,553 / 22,245,162 (99.665%)\n",
      "Discarded reads ...................: 0 / 22,245,162 (0.000%)\n",
      "Not assembled reads ...............: 74,609 / 22,245,162 (0.335%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SN89_R1_Plasmid_L1.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SN89_R1_Plasmid_L1.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SN89_R1_Plasmid_L1.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SN89_R1_Plasmid_L1.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SN89_R1_Plasmid_S5_L002_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SN89_R1_Plasmid_S5_L002_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.246601\n",
      "  C: 0.244655\n",
      "  G: 0.254125\n",
      "  T: 0.254619\n",
      "  49425 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 22,151,217 / 22,226,027 (99.663%)\n",
      "Discarded reads ...................: 0 / 22,226,027 (0.000%)\n",
      "Not assembled reads ...............: 74,810 / 22,226,027 (0.337%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SN89_R1_Plasmid_L2.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SN89_R1_Plasmid_L2.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SN89_R1_Plasmid_L2.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SN89_R1_Plasmid_L2.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SN89_R2_Plasmid_S6_L001_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SN89_R2_Plasmid_S6_L001_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.251579\n",
      "  C: 0.238804\n",
      "  G: 0.249375\n",
      "  T: 0.260242\n",
      "  50590 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 21,232,051 / 21,279,628 (99.776%)\n",
      "Discarded reads ...................: 0 / 21,279,628 (0.000%)\n",
      "Not assembled reads ...............: 47,577 / 21,279,628 (0.224%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SN89_R2_Plasmid_L1.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SN89_R2_Plasmid_L1.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SN89_R2_Plasmid_L1.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SN89_R2_Plasmid_L1.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SN89_R2_Plasmid_S6_L002_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SN89_R2_Plasmid_S6_L002_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.251504\n",
      "  C: 0.238846\n",
      "  G: 0.249405\n",
      "  T: 0.260246\n",
      "  46380 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 20,934,808 / 20,981,649 (99.777%)\n",
      "Discarded reads ...................: 0 / 20,981,649 (0.000%)\n",
      "Not assembled reads ...............: 46,841 / 20,981,649 (0.223%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SN89_R2_Plasmid_L2.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SN89_R2_Plasmid_L2.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SN89_R2_Plasmid_L2.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SN89_R2_Plasmid_L2.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SN89_R1_P1_S7_L001_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SN89_R1_P1_S7_L001_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.256856\n",
      "  C: 0.248112\n",
      "  G: 0.239381\n",
      "  T: 0.255651\n",
      "  56971 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 24,040,849 / 24,183,097 (99.412%)\n",
      "Discarded reads ...................: 0 / 24,183,097 (0.000%)\n",
      "Not assembled reads ...............: 142,248 / 24,183,097 (0.588%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SN89_R1_P1_L1.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SN89_R1_P1_L1.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SN89_R1_P1_L1.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SN89_R1_P1_L1.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SN89_R1_P1_S7_L002_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SN89_R1_P1_S7_L002_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.256791\n",
      "  C: 0.248151\n",
      "  G: 0.239406\n",
      "  T: 0.255652\n",
      "  53327 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 23,890,410 / 24,032,097 (99.410%)\n",
      "Discarded reads ...................: 0 / 24,032,097 (0.000%)\n",
      "Not assembled reads ...............: 141,687 / 24,032,097 (0.590%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SN89_R1_P1_L2.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SN89_R1_P1_L2.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SN89_R1_P1_L2.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SN89_R1_P1_L2.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SN89_R2_P1_S8_L001_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SN89_R2_P1_S8_L001_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.257024\n",
      "  C: 0.243344\n",
      "  G: 0.253456\n",
      "  T: 0.246176\n",
      "  51932 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 21,698,923 / 21,723,297 (99.888%)\n",
      "Discarded reads ...................: 0 / 21,723,297 (0.000%)\n",
      "Not assembled reads ...............: 24,374 / 21,723,297 (0.112%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SN89_R2_P1_L1.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SN89_R2_P1_L1.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SN89_R2_P1_L1.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SN89_R2_P1_L1.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SN89_R2_P1_S8_L002_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SN89_R2_P1_S8_L002_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.256890\n",
      "  C: 0.243449\n",
      "  G: 0.253494\n",
      "  T: 0.246168\n",
      "  49998 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 22,509,294 / 22,534,341 (99.889%)\n",
      "Discarded reads ...................: 0 / 22,534,341 (0.000%)\n",
      "Not assembled reads ...............: 25,047 / 22,534,341 (0.111%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SN89_R2_P1_L2.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SN89_R2_P1_L2.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SN89_R2_P1_L2.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SN89_R2_P1_L2.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SI06_R1_Plasmid_S9_L001_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SI06_R1_Plasmid_S9_L001_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.251492\n",
      "  C: 0.253704\n",
      "  G: 0.253808\n",
      "  T: 0.240996\n",
      "  31614 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 13,466,117 / 13,490,384 (99.820%)\n",
      "Discarded reads ...................: 0 / 13,490,384 (0.000%)\n",
      "Not assembled reads ...............: 24,267 / 13,490,384 (0.180%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SI06_R1_Plasmid_L1.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SI06_R1_Plasmid_L1.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SI06_R1_Plasmid_L1.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SI06_R1_Plasmid_L1.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SI06_R1_Plasmid_S9_L002_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SI06_R1_Plasmid_S9_L002_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.251411\n",
      "  C: 0.253774\n",
      "  G: 0.253799\n",
      "  T: 0.241016\n",
      "  31124 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 13,863,049 / 13,888,043 (99.820%)\n",
      "Discarded reads ...................: 0 / 13,888,043 (0.000%)\n",
      "Not assembled reads ...............: 24,994 / 13,888,043 (0.180%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SI06_R1_Plasmid_L2.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SI06_R1_Plasmid_L2.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SI06_R1_Plasmid_L2.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SI06_R1_Plasmid_L2.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SI06_R2_Plasmid_S10_L001_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SI06_R2_Plasmid_S10_L001_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.251412\n",
      "  C: 0.257870\n",
      "  G: 0.249921\n",
      "  T: 0.240796\n",
      "  46238 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 19,027,123 / 19,255,825 (98.812%)\n",
      "Discarded reads ...................: 0 / 19,255,825 (0.000%)\n",
      "Not assembled reads ...............: 228,702 / 19,255,825 (1.188%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SI06_R2_Plasmid_L1.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SI06_R2_Plasmid_L1.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SI06_R2_Plasmid_L1.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SI06_R2_Plasmid_L1.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SI06_R2_Plasmid_S10_L002_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SI06_R2_Plasmid_S10_L002_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.251341\n",
      "  C: 0.257896\n",
      "  G: 0.250002\n",
      "  T: 0.240761\n",
      "  44207 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 19,792,969 / 20,030,496 (98.814%)\n",
      "Discarded reads ...................: 0 / 20,030,496 (0.000%)\n",
      "Not assembled reads ...............: 237,527 / 20,030,496 (1.186%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SI06_R2_Plasmid_L2.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SI06_R2_Plasmid_L2.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SI06_R2_Plasmid_L2.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SI06_R2_Plasmid_L2.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SI06_R1_P1_S11_L001_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SI06_R1_P1_S11_L001_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.241197\n",
      "  C: 0.249384\n",
      "  G: 0.269746\n",
      "  T: 0.239673\n",
      "  46821 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 19,567,286 / 19,581,911 (99.925%)\n",
      "Discarded reads ...................: 0 / 19,581,911 (0.000%)\n",
      "Not assembled reads ...............: 14,625 / 19,581,911 (0.075%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SI06_R1_P1_L1.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SI06_R1_P1_L1.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SI06_R1_P1_L1.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SI06_R1_P1_L1.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SI06_R1_P1_S11_L002_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SI06_R1_P1_S11_L002_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.241197\n",
      "  C: 0.249328\n",
      "  G: 0.269883\n",
      "  T: 0.239592\n",
      "  43782 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 19,511,861 / 19,526,096 (99.927%)\n",
      "Discarded reads ...................: 0 / 19,526,096 (0.000%)\n",
      "Not assembled reads ...............: 14,235 / 19,526,096 (0.073%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SI06_R1_P1_L2.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SI06_R1_P1_L2.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SI06_R1_P1_L2.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SI06_R1_P1_L2.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SI06_R2_P1_S12_L001_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SI06_R2_P1_S12_L001_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.254665\n",
      "  C: 0.250630\n",
      "  G: 0.246473\n",
      "  T: 0.248233\n",
      "  45414 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 19,222,432 / 19,244,023 (99.888%)\n",
      "Discarded reads ...................: 0 / 19,244,023 (0.000%)\n",
      "Not assembled reads ...............: 21,591 / 19,244,023 (0.112%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SI06_R2_P1_L1.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SI06_R2_P1_L1.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SI06_R2_P1_L1.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SI06_R2_P1_L1.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SI06_R2_P1_S12_L002_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SI06_R2_P1_S12_L002_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.254536\n",
      "  C: 0.250699\n",
      "  G: 0.246609\n",
      "  T: 0.248156\n",
      "  42410 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 19,011,350 / 19,032,729 (99.888%)\n",
      "Discarded reads ...................: 0 / 19,032,729 (0.000%)\n",
      "Not assembled reads ...............: 21,379 / 19,032,729 (0.112%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SI06_R2_P1_L2.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SI06_R2_P1_L2.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SI06_R2_P1_L2.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SI06_R2_P1_L2.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SYD21_R1_Plasmid_S13_L001_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SYD21_R1_Plasmid_S13_L001_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.262114\n",
      "  C: 0.244344\n",
      "  G: 0.233210\n",
      "  T: 0.260331\n",
      "  39915 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 16,992,351 / 17,067,571 (99.559%)\n",
      "Discarded reads ...................: 0 / 17,067,571 (0.000%)\n",
      "Not assembled reads ...............: 75,220 / 17,067,571 (0.441%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SYD21_R1_Plasmid_L1.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SYD21_R1_Plasmid_L1.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SYD21_R1_Plasmid_L1.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SYD21_R1_Plasmid_L1.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SYD21_R1_Plasmid_S13_L002_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SYD21_R1_Plasmid_S13_L002_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.261966\n",
      "  C: 0.244471\n",
      "  G: 0.233232\n",
      "  T: 0.260330\n",
      "  38999 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 17,419,113 / 17,496,045 (99.560%)\n",
      "Discarded reads ...................: 0 / 17,496,045 (0.000%)\n",
      "Not assembled reads ...............: 76,932 / 17,496,045 (0.440%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SYD21_R1_Plasmid_L2.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SYD21_R1_Plasmid_L2.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SYD21_R1_Plasmid_L2.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SYD21_R1_Plasmid_L2.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SYD21_R2_Plasmid_S14_L001_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SYD21_R2_Plasmid_S14_L001_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.261290\n",
      "  C: 0.239420\n",
      "  G: 0.239138\n",
      "  T: 0.260151\n",
      "  39196 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 16,378,556 / 16,397,178 (99.886%)\n",
      "Discarded reads ...................: 0 / 16,397,178 (0.000%)\n",
      "Not assembled reads ...............: 18,622 / 16,397,178 (0.114%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SYD21_R2_Plasmid_L1.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SYD21_R2_Plasmid_L1.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SYD21_R2_Plasmid_L1.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SYD21_R2_Plasmid_L1.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SYD21_R2_Plasmid_S14_L002_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SYD21_R2_Plasmid_S14_L002_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.261213\n",
      "  C: 0.239489\n",
      "  G: 0.239113\n",
      "  T: 0.260185\n",
      "  37470 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 16,537,339 / 16,555,905 (99.888%)\n",
      "Discarded reads ...................: 0 / 16,555,905 (0.000%)\n",
      "Not assembled reads ...............: 18,566 / 16,555,905 (0.112%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SYD21_R2_Plasmid_L2.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SYD21_R2_Plasmid_L2.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SYD21_R2_Plasmid_L2.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SYD21_R2_Plasmid_L2.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SYD21_R1_P1_S15_L001_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SYD21_R1_P1_S15_L001_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.250225\n",
      "  C: 0.256770\n",
      "  G: 0.244674\n",
      "  T: 0.248331\n",
      "  48359 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 20,274,531 / 20,322,554 (99.764%)\n",
      "Discarded reads ...................: 0 / 20,322,554 (0.000%)\n",
      "Not assembled reads ...............: 48,023 / 20,322,554 (0.236%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SYD21_R1_P1_L1.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SYD21_R1_P1_L1.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SYD21_R1_P1_L1.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SYD21_R1_P1_L1.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SYD21_R1_P1_S15_L002_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SYD21_R1_P1_S15_L002_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.250118\n",
      "  C: 0.256862\n",
      "  G: 0.244708\n",
      "  T: 0.248312\n",
      "  44380 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 20,154,534 / 20,202,124 (99.764%)\n",
      "Discarded reads ...................: 0 / 20,202,124 (0.000%)\n",
      "Not assembled reads ...............: 47,590 / 20,202,124 (0.236%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SYD21_R1_P1_L2.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SYD21_R1_P1_L2.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SYD21_R1_P1_L2.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SYD21_R1_P1_L2.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SYD21_R2_P1_S16_L001_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SYD21_R2_P1_S16_L001_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.264914\n",
      "  C: 0.239444\n",
      "  G: 0.236632\n",
      "  T: 0.259010\n",
      "  55001 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 23,639,285 / 23,990,509 (98.536%)\n",
      "Discarded reads ...................: 0 / 23,990,509 (0.000%)\n",
      "Not assembled reads ...............: 351,224 / 23,990,509 (1.464%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SYD21_R2_P1_L1.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SYD21_R2_P1_L1.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SYD21_R2_P1_L1.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SYD21_R2_P1_L1.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: data/sequencing_data/SYD21_R2_P1_S16_L002_R1_001.fastq.gz\n",
      "Reverse reads file.................: data/sequencing_data/SYD21_R2_P1_S16_L002_R2_001.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 6\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.264870\n",
      "  C: 0.239447\n",
      "  G: 0.236732\n",
      "  T: 0.258951\n",
      "  53525 uncalled bases\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 23,826,645 / 24,182,391 (98.529%)\n",
      "Discarded reads ...................: 0 / 24,182,391 (0.000%)\n",
      "Not assembled reads ...............: 355,746 / 24,182,391 (1.471%)\n",
      "Assembled reads file...............: data/sequencing_data/pear_output/SYD21_R2_P1_L2.assembled.fastq\n",
      "Discarded reads file...............: data/sequencing_data/pear_output/SYD21_R2_P1_L2.discarded.fastq\n",
      "Unassembled forward reads file.....: data/sequencing_data/pear_output/SYD21_R2_P1_L2.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: data/sequencing_data/pear_output/SYD21_R2_P1_L2.unassembled.reverse.fastq\n"
     ]
    }
   ],
   "source": [
    "#for idx,row in samples.iterrows():\n",
    "#    sample_name = row['sample_name']\n",
    "#    fwd_read = row['read_1_file']\n",
    "#    rev_read = row['read_2_file']\n",
    "#    os.system(f'pear -f {fwd_read} -r {rev_read} -o \"data/sequencing_data/pear_output/\"{sample_name} -j 6')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "76773e59-6234-49b5-bc90-6085bc803221",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "17891620it [03:25, 87021.77it/s]\n",
      "18481616it [03:35, 85608.60it/s]\n",
      "18553528it [03:24, 90775.42it/s]\n",
      "19202185it [03:30, 91305.03it/s]\n",
      "27430342it [05:21, 85375.19it/s]\n",
      "27162240it [05:22, 84264.72it/s]\n",
      "21998618it [04:18, 84997.17it/s]\n",
      "21895884it [04:18, 84853.67it/s]\n",
      "22170553it [04:25, 83592.19it/s]\n",
      "22151217it [04:24, 83771.33it/s]\n",
      "21232051it [04:15, 83044.41it/s]\n",
      "20934808it [04:10, 83421.05it/s]\n",
      "24040849it [04:35, 87298.99it/s]\n",
      "23890410it [04:36, 86528.74it/s]\n",
      "21698923it [04:11, 86225.11it/s]\n",
      "22509294it [04:17, 87449.80it/s]\n",
      "13466117it [02:43, 82601.77it/s]\n",
      "13863049it [02:48, 82236.21it/s]\n",
      "19027123it [03:50, 82459.52it/s]\n",
      "19792969it [03:57, 83283.32it/s]\n",
      "19567286it [03:43, 87559.13it/s]\n",
      "19511861it [03:43, 87108.52it/s]\n",
      "19222432it [03:42, 86262.40it/s]\n",
      "19011350it [03:43, 85098.24it/s]\n",
      "16992351it [03:27, 81867.88it/s]\n",
      "17419113it [03:30, 82808.83it/s]\n",
      "16378556it [03:15, 83800.00it/s]\n",
      "16537339it [03:17, 83861.53it/s]\n",
      "20274531it [03:34, 94527.63it/s]\n",
      "20154534it [03:33, 94332.85it/s]\n",
      "23639285it [04:31, 86928.92it/s]\n",
      "23826645it [04:37, 85996.42it/s]\n"
     ]
    }
   ],
   "source": [
    "flanks=['AGCCTCCAGTGTAGGATTTGCATTTAA','TGGTGCTTTTGGTCTCCCTGGGG']\n",
    "all_data = []\n",
    "for index,row in samples.iterrows():\n",
    "    gc.collect()\n",
    "    sample_name = row['sample_name']\n",
    "    virus_background = row['virus_background']\n",
    "    replicate = row['replicate']\n",
    "    selection = row['selection']\n",
    "    inline_fwd = row['inline_fwd']\n",
    "    inline_rev = row['inline_rev']\n",
    "    inline_rev_rcomp = str(Seq(inline_rev).reverse_complement())\n",
    "    required_length = 78 + 16 + len(inline_fwd) + len(inline_rev)\n",
    "    barcode_umi_counts = dict()\n",
    "    for record in tqdm.tqdm(SeqIO.parse(f'data/sequencing_data/pear_output/{sample_name}.assembled.fastq', 'fastq')):\n",
    "        seq_str = str(record.seq)\n",
    "        if len(seq_str) == required_length and 'N' not in seq_str and inline_fwd in seq_str[8:8+len(inline_fwd)] and inline_rev_rcomp in seq_str[-8-len(inline_rev_rcomp):-8]:\n",
    "            UMI_merged = seq_str[:8]+seq_str[-8:]\n",
    "            if flanks[0] in seq_str and flanks[1] in seq_str:\n",
    "                barcode = seq_str.split(flanks[0])[1].split(flanks[1])[0]                   \n",
    "            if flanks[0] in seq_str and flanks[1] not in seq_str:\n",
    "                barcode = seq_str.split(flanks[0])[1][:28]\n",
    "            if flanks[0] not in seq_str and flanks[1] in seq_str:\n",
    "                barcode = seq_str.split(flanks[1])[0][-28:]\n",
    "                \n",
    "            barcode_umi_counts.setdefault(barcode, dict())\n",
    "            barcode_umi_counts[barcode].setdefault(UMI_merged, 0)\n",
    "            barcode_umi_counts[barcode][UMI_merged] += 1\n",
    "            \n",
    "    data = {'barcode':[], 'umi':[], 'count':[]}\n",
    "    for barcode, umis in barcode_umi_counts.items():\n",
    "        for umi, count in umis.items():\n",
    "            data['barcode'].append(barcode)\n",
    "            data['umi'].append(umi)\n",
    "            data['count'].append(count)\n",
    "    data = pd.DataFrame(data)\n",
    "    #data['sample_name'] = row['sample_name']\n",
    "    data['sample'] = row['virus_background']+'_'+row['replicate']+'_'+row['selection']\n",
    "    data['virus_background'] = row['virus_background']\n",
    "    data['replicate'] = row['replicate']\n",
    "    data['selection'] = row['selection']\n",
    "    \n",
    "    data.to_csv(f'data/barcode_umi_counts/{sample_name}_barcodes_umis.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "id": "322b461b-8ccb-46f8-aa75-f012f2a535b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "sample_list=[]\n",
    "for idx,row in samples.iterrows():\n",
    "    sample=row['sample_name'][:-3]\n",
    "    if sample in sample_list:\n",
    "        continue\n",
    "    if sample not in sample_list:\n",
    "        sample_list.append(sample)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "id": "8c5c746b-87f5-49e6-856d-9bfff9158bc3",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆ| 16/16 [17:10<00:00, 64.38s/it]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>virus_background</th>\n",
       "      <th>replicate</th>\n",
       "      <th>selection</th>\n",
       "      <th>sample</th>\n",
       "      <th>barcode</th>\n",
       "      <th>bc_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>USSR77</td>\n",
       "      <td>R1</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>USSR77_R1_Plasmid</td>\n",
       "      <td>AAAACACGCACAAGCTAATACGAAACGA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>USSR77</td>\n",
       "      <td>R1</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>USSR77_R1_Plasmid</td>\n",
       "      <td>AAAACGACTGCTAGCTCCTAACCGTGGT</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>USSR77</td>\n",
       "      <td>R1</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>USSR77_R1_Plasmid</td>\n",
       "      <td>AAAACTCACTTAAGCTCGGCCCAAGGAT</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>USSR77</td>\n",
       "      <td>R1</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>USSR77_R1_Plasmid</td>\n",
       "      <td>AAAAGGTGTCTTAGCTTGTTCATTATCC</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>USSR77</td>\n",
       "      <td>R1</td>\n",
       "      <td>Plasmid</td>\n",
       "      <td>USSR77_R1_Plasmid</td>\n",
       "      <td>AAAAGTTGACATAGCTGGTCAATATTAC</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10193604</th>\n",
       "      <td>SYD21</td>\n",
       "      <td>R2</td>\n",
       "      <td>P1</td>\n",
       "      <td>SYD21_R2_P1</td>\n",
       "      <td>TTTTGCTAGGGTAGCTTGTATCACGTGG</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10193605</th>\n",
       "      <td>SYD21</td>\n",
       "      <td>R2</td>\n",
       "      <td>P1</td>\n",
       "      <td>SYD21_R2_P1</td>\n",
       "      <td>TTTTGGGACTACAGCTATCTCATCTCAG</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10193606</th>\n",
       "      <td>SYD21</td>\n",
       "      <td>R2</td>\n",
       "      <td>P1</td>\n",
       "      <td>SYD21_R2_P1</td>\n",
       "      <td>TTTTTGACCTCATGCTACCTTGTATTCA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10193607</th>\n",
       "      <td>SYD21</td>\n",
       "      <td>R2</td>\n",
       "      <td>P1</td>\n",
       "      <td>SYD21_R2_P1</td>\n",
       "      <td>TTTTTTATTGTTAGCTTATAAACTGGTA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10193608</th>\n",
       "      <td>SYD21</td>\n",
       "      <td>R2</td>\n",
       "      <td>P1</td>\n",
       "      <td>SYD21_R2_P1</td>\n",
       "      <td>TTTTTTTACGTCAGCTTCGATTTCACCT</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10193609 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         virus_background replicate selection             sample  \\\n",
       "0                  USSR77        R1   Plasmid  USSR77_R1_Plasmid   \n",
       "1                  USSR77        R1   Plasmid  USSR77_R1_Plasmid   \n",
       "2                  USSR77        R1   Plasmid  USSR77_R1_Plasmid   \n",
       "3                  USSR77        R1   Plasmid  USSR77_R1_Plasmid   \n",
       "4                  USSR77        R1   Plasmid  USSR77_R1_Plasmid   \n",
       "...                   ...       ...       ...                ...   \n",
       "10193604            SYD21        R2        P1        SYD21_R2_P1   \n",
       "10193605            SYD21        R2        P1        SYD21_R2_P1   \n",
       "10193606            SYD21        R2        P1        SYD21_R2_P1   \n",
       "10193607            SYD21        R2        P1        SYD21_R2_P1   \n",
       "10193608            SYD21        R2        P1        SYD21_R2_P1   \n",
       "\n",
       "                               barcode  bc_count  \n",
       "0         AAAACACGCACAAGCTAATACGAAACGA         1  \n",
       "1         AAAACGACTGCTAGCTCCTAACCGTGGT         1  \n",
       "2         AAAACTCACTTAAGCTCGGCCCAAGGAT         1  \n",
       "3         AAAAGGTGTCTTAGCTTGTTCATTATCC         1  \n",
       "4         AAAAGTTGACATAGCTGGTCAATATTAC         1  \n",
       "...                                ...       ...  \n",
       "10193604  TTTTGCTAGGGTAGCTTGTATCACGTGG         1  \n",
       "10193605  TTTTGGGACTACAGCTATCTCATCTCAG         1  \n",
       "10193606  TTTTTGACCTCATGCTACCTTGTATTCA         1  \n",
       "10193607  TTTTTTATTGTTAGCTTATAAACTGGTA         1  \n",
       "10193608  TTTTTTTACGTCAGCTTCGATTTCACCT         1  \n",
       "\n",
       "[10193609 rows x 6 columns]"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_data_df=pd.DataFrame()\n",
    "\n",
    "for f in tqdm.tqdm(sample_list):\n",
    "    gc.collect()\n",
    "    file_lane1 = f+'_L1_barcodes_umis.csv'\n",
    "    file_lane2 = f+'_L2_barcodes_umis.csv'\n",
    "    \n",
    "    if file_lane1 and file_lane2 in os.listdir('data/barcode_umi_counts'):  \n",
    "        temp_df1 = pd.read_csv(f'data/barcode_umi_counts/{file_lane1}')\n",
    "        temp_df2 = pd.read_csv(f'data/barcode_umi_counts/{file_lane2}')\n",
    "        temp_df = pd.concat([temp_df1,temp_df2])\n",
    "        temp_df['barcode_umi']=temp_df['barcode']+'_'+temp_df['umi']\n",
    "        temp_df = temp_df.drop_duplicates(subset=['barcode_umi','sample'],keep='first')\n",
    "        temp_df['bc_count']=1\n",
    "        temp_df_agg = temp_df.groupby(['virus_background','replicate','selection','sample','barcode']).agg({'bc_count':'sum'}).reset_index()\n",
    "        all_data_df=pd.concat([all_data_df,temp_df_agg]).reset_index(drop=True)\n",
    "    \n",
    "    elif file_lane1 in os.listdir('data/barcode_umi_counts') and file_lane2 not in os.listdir('data/barcode_umi_counts'):  \n",
    "        temp_df = pd.read_csv(f'data/barcode_umi_counts/{file_lane1}')\n",
    "        temp_df['bc_count']=1\n",
    "        temp_df['barcode_umi']=temp_df['barcode']+'_'+temp_df['umi']\n",
    "        temp_df_agg = temp_df.groupby(['virus_background','replicate','selection','sample','barcode']).agg({'bc_count':'sum'}).reset_index()\n",
    "        all_data_df=pd.concat([all_data_df,temp_df_agg]).reset_index(drop=True)\n",
    "        \n",
    "    elif file_lane2 in os.listdir('data/barcode_umi_counts') and file_lane1 not in os.listdir('data/barcode_umi_counts'):  \n",
    "        temp_df = pd.read_csv(f'data/barcode_umi_counts/{file_lane2}')\n",
    "        temp_df['bc_count']=1\n",
    "        temp_df['barcode_umi']=temp_df['barcode']+'_'+temp_df['umi']\n",
    "        temp_df_agg = temp_df.groupby(['virus_background','replicate','selection','sample','barcode']).agg({'bc_count':'sum'}).reset_index()\n",
    "        all_data_df=pd.concat([all_data_df,temp_df_agg]).reset_index(drop=True)\n",
    "    \n",
    "    else:\n",
    "        continue\n",
    "\n",
    "all_data_df.to_csv('data/all_data_df.csv',index=False)\n",
    "all_data_df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "flu_selections",
   "language": "python",
   "name": "flu_selections"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.17"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
